cupy/cuda/memory.pyx

```diff
@@ -540,6 +540,30 @@ cpdef MemoryPointer alloc(size):
     return _current_allocator(size)
 
 
+cdef class _ActiveAllocator(object):
+
+    """ Context manager for setting an allocator
+
+    An object of this class is returned by set_allocator.
+    This allows it to work as a regular function and a
+    context manager
+
+    """
+
+    cdef:
+        readonly object p_allocator
```

Comment:
Please rename this variable to `_p_allocator` because it is a private attribute.